cleanup circle config

zarsky-broad opened the pull request.

zarsky-broad committed:
cleanup circle config

zarsky-broad committed:
add step label back

panentheos commented on .circleci/config.yml:
```diff
@@ -7,12 +7,14 @@ jobs:
+          key: deps2-{{ checksum "package.json" }}
           paths:
-            - node_modules
+            - ~/.npm
```
I'm curious why this is preferred over caching the whole node_modules dir?

zarsky-broad replied:
In going back to get the link to my source for this, I think I misread it. I'll change this back.

panentheos reviewed (approved):
Looks good.

zarsky-broad committed:
recommended way

based on https://github.com/CircleCI-Public/circleci-demo-javascript-express/blob/master/.circleci/config.yml
